### PR TITLE
fix(contracts): Remove ambiguity in results value

### PR DESCRIPTION
**Pull request by mfornet**

### CODE REVIEW

File: erc20-connector/contracts/BridgeTokenFactory.sol [hunk 1]
```diff
@@ -79,6 +79,8 @@ contract BridgeTokenFactory is ERC20Locker {
 
     function _decodeLockResult(bytes memory data) internal pure returns(LockResult memory result) {
         Borsh.Data memory borshData = Borsh.from(data);
+        uint8 memory flag = borshData.decodeU8();
```
Comment (abacabadabacaba) [inline erc20-connector/contracts/BridgeTokenFactory.sol:82]: Remove `memory`, `uint8` is a value type so it doesn't require storage specifier.

File: erc20-connector/contracts/ERC20Locker.sol [hunk 2]
```diff
@@ -49,6 +49,8 @@ contract ERC20Locker is Locker {
 
     function _decodeBurnResult(bytes memory data) internal pure returns(BurnResult memory result) {
         Borsh.Data memory borshData = Borsh.from(data);
+        uint8 memory flag = borshData.decodeU8();
```
Comment (abacabadabacaba) [inline erc20-connector/contracts/ERC20Locker.sol:52]: Same here.

File: erc20-connector/contracts/BridgeTokenFactory.sol (hunk 1 above)
Comment (mfornet) [reply erc20-connector/contracts/BridgeTokenFactory.sol:82]: 🆗 

I still don't understand how is memory managed in solidity, I used `memory` from my understanding after reading here:
https://solidity.readthedocs.io/en/develop/introduction-to-smart-contracts.html#storage-memory-and-the-stack

So from you comment I understand that it is similar to rust traits `Copy` and `Clone`, in this case some basic types (such as `uint8`) is `Copy` so its value is not stored in `storage` but instead is directly stored in `memory`, without needing to specify it.

**Review by abacabadabacaba — APPROVED** [review]

File: bridge-token-factory/src/lib.rs [hunk 3]
```diff
@@ -33,6 +33,12 @@ const TRANSFER_FROM_GAS: Gas = 10_000_000_000_000;
 
 const TRANSFER_GAS: Gas = 10_000_000_000_000;
 
+#[derive(Debug, Eq, PartialEq, BorshSerialize, BorshDeserialize)]
+pub enum ResultType {
```
Comment (nearmax) [inline bridge-token-factory/src/lib.rs:37]: I would rather make it
```rust
pub enum ResultType {
  Withdraw(u128, [u8; 20], [u8; 20]),
  Lock(String, u128, [u8; 20]),
}
```
but I guess it is a matter of taste :)

File: erc20-connector/dist/BridgeTokenFactory.full.sol [hunk 4]
```diff
@@ -529,7 +529,7 @@ library Borsh {
         }
     }
 
-    function decodeBytes20(Data memory data) internal pure returns(bytes20 value) {
+    function decodeBytes20(Data memory data) internal pure shift(data, 20) returns(bytes20 value) {
```
Comment (nearmax) [inline erc20-connector/dist/BridgeTokenFactory.full.sol:532]: I wonder where did `shift(data, 20)` come from? This looks suspicious. @abacabadabacaba WDYT?

**Review by nearmax — APPROVED** [review]

**Pull request merged**